packages/protocol/contracts/stability/StableToken.sol

```diff
@@ -461,47 +462,74 @@ contract StableToken is
   }
 
   /**
-   * @notice Deduct balance for making payments for gas in this StableToken currency.
-   * @param from The account to debit balance from
-   * @param value The value of balance to debit
+   * @notice Transfers StableToken from one address to another
+   * @param to The address to transfer StableToken to.
+   * @param value The amount of StableToken to be transferred.
+   */
+  function _transfer(address to, uint256 value) internal returns (bool) {
+    require(to != address(0), "transfer attempted to reserved address 0x0");
+    uint256 units = _valueToUnits(inflationState.factor, value);
+    require(balances[msg.sender] >= units, "transfer value exceeded balance of sender");
+    balances[msg.sender] = balances[msg.sender].sub(units);
+    balances[to] = balances[to].add(units);
+    emit Transfer(msg.sender, to, value);
+    return true;
+  }
+
+  /**
+   * @notice Reserve balance for making payments for gas in this StableToken currency.
+   * @param from The account to reserve balance from
+   * @param value The amount of balance to reserve
    */
-  function debitFrom(address from, uint256 value)
+  function debitGasFees(address from, uint256 value)
     external
     onlyVm
-    updateInflationFactor
     onlyWhenNotFrozen
+    updateInflationFactor
   {
     uint256 units = _valueToUnits(inflationState.factor, value);
-    totalSupply_ = totalSupply_.sub(units);
     balances[from] = balances[from].sub(units);
+    totalSupply_ = totalSupply_.sub(units);
   }
 
   /**
-   * @notice Refund balance after making payments for gas in this StableToken currency.
-   * @param to The account to credit balance to
-   * @param value The amount of balance to credit
-   * @dev We can assume that the inflation factor is up to date as `debitFrom`
-   * will have been called in the same transaction
+   * @notice Alternative function to credit balance after making payments
+   * for gas in this StableToken currency.
+   * @param from The account to debit balance from
+   * @param feeRecipient Coinbase address
+   * @param gatewayFeeRecipient Gateway address
+   * @param communityFund Community fund address
+   * @param tipTxFee Coinbase fee
+   * @param baseTxFee Community fund fee
+   * @param gatewayFee Gateway fee
    */
-  function creditTo(address to, uint256 value) external onlyVm onlyWhenNotFrozen {
-    uint256 units = _valueToUnits(inflationState.factor, value);
+  function creditGasFees(
+    address from,
+    address feeRecipient,
+    address gatewayFeeRecipient,
+    address communityFund,
+    uint256 refund,
+    uint256 tipTxFee,
+    uint256 gatewayFee,
+    uint256 baseTxFee
+  ) external onlyVm onlyWhenNotFrozen {
+    uint256 units = _valueToUnits(inflationState.factor, refund);
+    balances[from] = balances[from].add(units);
```

Comment:
this should be the last line... since creditGas does not decrement the balance...

Reply:
The refund is computed in the celo-blockchain now.

Reply:
nevermind... i was reading the previous impl in my head